examples/basics/demo_physics.py

```diff
@@ -0,0 +1,165 @@
+# ----------------------------------------------------------------------------------------
+# Defining a new linear operator only requires a forward function and its transpose operation,
+# inheriting the remaining structure of the :class:`deepinv.physics.LinearPhysics` class.
+#
+# Once the operator is defined, we can use any of the functions in the :class:`deepinv.physics.Physics` class and
+# :class:`deepinv.physics.LinearPhysics` class, such as computing the norm of the operator, testing the adjointness,
+# computing the proximal operator, etc.
+
+
+# define an operator that converts color images into grayscale ones.
+class Decolorize(dinv.physics.LinearPhysics):
+    r"""
+    Converts RGB images to grayscale.
+
+    Signals must be tensors with 3 colour (RGB) channels, i.e. [*,3,*,*]
+    The measurements are grayscale images.
+
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.noise_model = dinv.physics.GaussianNoise(sigma=0.1)
+
+    def A(self, x):
+        y = x[:, 0, :, :] * 0.2989 + x[:, 1, :, :] * 0.5870 + x[:, 2, :, :] * 0.1140
+        return y.unsqueeze(1)
+
+    def A_adjoint(self, y):
+        return torch.cat([y * 0.2989, y * 0.5870, y * 0.1140], dim=1)
+
+
+physics = Decolorize()
+
+# %%
+# Generate toy image
+# --------------------------------------------------------------------
+#
+# This example uses a toy image with 3 color channels.
+
+
+x = torch.zeros(1, 3, 96, 128)
+x[:, 0, :32, :] = 1
+x[:, 1, 32:64, :] = 1
+x[:, 2, 64:, :] = 1
+
+y = physics(x)
+xlin = physics.A_dagger(y)  # compute the linear pseudo-inverse
+
+dinv.utils.plot(
+    [x, y, xlin],
+    titles=["image", "meas.", "linear rec."],
+)
+
+
+# %%
+# Verifying our linear operator
+# --------------------------------------------
+#
+#     If the operator is linear, it is recommended to verify that the transpose well-defined using
+#     :meth:`deepinv.physics.LinearPhysics.adjointness_test()`,
+#     and that it has a unit norm using :meth:`deepinv.physics.LinearPhysics.compute_norm()`.
+
+print(f"The linear operator has norm={physics.compute_norm(x):.2f}")
+
+if physics.adjointness_test(x) < 1e-5:
+    print("The linear operator has a well defined transpose")
+
+
+# %%
+# Creating a decomposable forward operator.
+# --------------------------------------------
+# If the forward operator has a closed form singular value decomposition (SVD),
+# it is recommended to implement the operator using the :meth:`deepinv.physics.DecomposablePhysics` method.
+#
+# The operator in this example is decomposable, so we can implement it using
+# :class:`deepinv.physics.DecomposablePhysics`.
+
+
+class DecolorizeSVD(dinv.physics.DecomposablePhysics):
+    r"""
+    Converts RGB images to grayscale.
+
+    Signals must be tensors with 3 colour (RGB) channels, i.e. [*,3,*,*]
+    The measurements are grayscale images.
+
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.mask = 0.447
```

Comment:
seems unused? It is not clear what you need this so this should be explained in the example.